lib/puppet/pops/parser/slurp_support.rb

```diff
@@ -29,10 +31,18 @@ def slurp_sqstring
     str[0..-2] # strip closing "'" from result
   end
 
+  def slurp_btstring
+    # skip the leading `
+    @scanner.pos += 1
+    str = @scanner.scan_until(SLURP_BT_PATTERN)
+    lex_error(Issues::UNCLOSED_QUOTE, :after => "\"`\"", :followed_by => followed_by) unless str
+    str[0..-2] # strip closing "`" from result
+  end
+
   def slurp_dqstring
     scn = @scanner
     last = scn.matched
-    str = slurp(scn, SLURP_DQ_PATTERN, DQ_ESCAPES, false)
+    str = slurp(scn, SLURP_DQ_PATTERN, @handle_hex_escapes ?  DQ_ESCAPES_WITH_HEX : DQ_ESCAPES, false)
```

Comment:
This takes place per dq string - can you get rid of the extra conditional here? - i.e. set an instance var to either DQ_ESCAPES_WITH_HEX or DQ_ESCAPES and then access that?